src/services/api.service.ts

```diff
@@ -150,6 +133,45 @@ export const getSeriesByMediaIds = async (mediaIds: string[]): Promise<{ [key in
   return await getDataOrThrow(response);
 };
 
+// Max count per request
+const MAX_EPISODES_COUNT = 50;
+
+/**
+ * Get all episodes of the selected series
+ * @param {string} seriesId
+ */
+export const getEpisodes = async (seriesId: string | undefined): Promise<{ [id: string]: PlaylistItem }> => {
+  if (!seriesId) {
+    throw new Error('Series ID is required');
+  }
+
+  const pathname = `/apps/series/${seriesId}/episodes`;
+  const url = addQueryParams(`${import.meta.env.APP_API_BASE_URL}${pathname}`, {});
+
+  const response = await fetch(url);
+  const data = await getDataOrThrow(response);
+
+  let media = data.media;
+
+  // We will send additional requests in case there are more than 50 items
+  if (data.total > MAX_EPISODES_COUNT) {
+    // How many requests do we need to make (excluding one initially done)
+    const chunks = Math.ceil(data.total / MAX_EPISODES_COUNT) - 1;
+    const fetchedData = await Promise.all(
+      // Starting from the second page
+      Array.from({ length: chunks }, (_, i) => i + 2).map((page) => {
+        return fetch(`${url}?page=${page}`).then((r) => getDataOrThrow(r));
+      }),
+    );
+    const processedData = fetchedData.reduce((acc, el) => ({ ...acc, ...el.data }), {});
+
+    media = { processedData };
+  }
+
+  // Adding images and keys for media items
+  return Object.keys(media).reduce((acc, key) => ({ ...acc, [key]: transformMediaItem({ ...media[key], mediaid: key }) }), {});
```

Comment:
It would also be nice if the backend went ahead and added the episode numbers and season numbers so we don't have to rebuild those on the FE each time (I think they're already available in the BE.) Then I don't think we'll even need to call the series metadata endpoint (since we also have title, description, and custom params from the media item instead now.)

Reply:
Here we add some images data for media items to support alternate images.
We can probably add episodes and seasons to bulk-episodes endpoint. I will create a ticket.